{
  "affected": [
    {
      "database_specific": {
        "source": "https://storage.googleapis.com/cve-osv-conversion/osv-output/CVE-2018-1061.json"
      },
      "ranges": [
        {
          "events": [
            {
              "introduced": "0"
            },
            {
              "fixed": "ca079a3ea30098aff3197c559a0e32d42dda6d84"
            }
          ],
          "repo": "https://github.com/python/cpython",
          "type": "GIT"
        }
      ],
      "versions": []
    }
  ],
  "aliases": [
    "PSF-2018-4"
  ],
  "details": "python before versions 2.7.15, 3.4.9, 3.5.6rc1, 3.6.5rc1 and 3.7.0 is vulnerable to catastrophic backtracking in the difflib.IS_LINE_JUNK method.  An attacker could use this flaw to cause denial of service.",
  "id": "CVE-2018-1061",
  "modified": "2025-10-14T14:33:14Z",
  "published": "2018-06-19T12:29:00Z",
  "references": [
    {
      "type": "ADVISORY",
      "url": "http://www.securitytracker.com/id/1042001"
    },
    {
      "type": "ADVISORY",
      "url": "https://access.redhat.com/errata/RHBA-2019:0327"
    },
    {
      "type": "ADVISORY",
      "url": "https://access.redhat.com/errata/RHSA-2018:3041"
    },
    {
      "type": "ADVISORY",
      "url": "https://access.redhat.com/errata/RHSA-2018:3505"
    },
    {
      "type": "ADVISORY",
      "url": "https://access.redhat.com/errata/RHSA-2019:1260"
    },
    {
      "type": "ADVISORY",
      "url": "https://access.redhat.com/errata/RHSA-2019:3725"
    },
    {
      "type": "ADVISORY",
      "url": "https://bugs.python.org/issue32981"
    },
    {
      "type": "ADVISORY",
      "url": "https://docs.python.org/3.5/whatsnew/changelog.html#python-3-5-6-release-candidate-1"
    },
    {
      "type": "ADVISORY",
      "url": "https://docs.python.org/3.6/whatsnew/changelog.html#python-3-6-5-release-candidate-1"
    },
    {
      "type": "ARTICLE",
      "url": "https://lists.debian.org/debian-lts-announce/2018/09/msg00030.html"
    },
    {
      "type": "ARTICLE",
      "url": "https://lists.debian.org/debian-lts-announce/2018/09/msg00031.html"
    },
    {
      "type": "ADVISORY",
      "url": "https://usn.ubuntu.com/3817-1/"
    },
    {
      "type": "ADVISORY",
      "url": "https://usn.ubuntu.com/3817-2/"
    },
    {
      "type": "ADVISORY",
      "url": "https://www.debian.org/security/2018/dsa-4306"
    },
    {
      "type": "ADVISORY",
      "url": "https://www.debian.org/security/2018/dsa-4307"
    },
    {
      "type": "REPORT",
      "url": "https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2018-1061"
    },
    {
      "type": "WEB",
      "url": "http://lists.opensuse.org/opensuse-security-announce/2020-01/msg00040.html"
    },
    {
      "type": "WEB",
      "url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/46PVWY5LFP4BRPG3BVQ5QEEFYBVEXHCK/"
    },
    {
      "type": "WEB",
      "url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/AEZ5IQT7OF7Q2NCGIVABOWYGKO7YU3NJ/"
    },
    {
      "type": "WEB",
      "url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/JSKPGPZQNTAULHW4UH63KGOOUIDE4RRB/"
    },
    {
      "type": "WEB",
      "url": "https://support.hpe.com/hpsc/doc/public/display?docLocale=en_US&docId=emr_na-hpesbst03951en_us"
    }
  ],
  "related": [
    "MGASA-2018-0256",
    "MGASA-2018-0270",
    "SUSE-SU-2018:2408-1",
    "SUSE-SU-2018:2696-1",
    "SUSE-SU-2018:3554-1",
    "SUSE-SU-2018:3554-2",
    "SUSE-SU-2020:0114-1",
    "SUSE-SU-2020:0234-1",
    "openSUSE-SU-2020:0086-1",
    "openSUSE-SU-2024:11202-1",
    "openSUSE-SU-2024:11284-1"
  ],
  "schema_version": "1.7.3",
  "severity": [
    {
      "score": "CVSS:3.0/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
      "type": "CVSS_V3"
    }
  ]
}